### DISCUSSION TITLE
Why is setting root.x not taking effect?

### DISCUSSION THREAD

**Discussion by daydaystudylab** [slint-ui/slint#4351] [Q&A]
I want to achieve the effect of dragging the window, but I can only move within the black box

```
export component App inherits Window {

        width: 890px;
        height: 624px;
        title: "demo";
        no-frame: true;
        background: #333;

        area := TouchArea {
            width: parent.width;
            height: parent.height;
            moved => {
                root.x += self.mouse-x - self.pressed-x;
                root.y += self.mouse-y - self.pressed-y;
                
            }
        }

        Rectangle {
            background: #eee;
            Text {
                text: "hello drag app demo";
            }
        }
    }
```

**Comment by Vadoola** [comment]
You can't directly relocate the window from within Slint. How / whether this can be done will also depend on the environment you are working in. For example Wayland, typically doesn't like allowing the windows to move themselves. There are some discussions on ways to make this work, but right now, it's something you would need to do in code, not from within the Slint DSL.

See previous issues and discussions below:

- https://github.com/slint-ui/slint/discussions/3355
- https://github.com/slint-ui/slint/discussions/697
- https://github.com/slint-ui/slint/issues/323
- https://github.com/slint-ui/slint/discussions/4226
- https://github.com/slint-ui/slint/issues/2695
- https://github.com/slint-ui/slint/issues/613
- https://github.com/slint-ui/slint/issues/2521

**Reply by daydaystudylab** [reply]
ok，thanks
